src/nodes/primitive/range.ts

```diff
@@ -63,17 +91,24 @@ export type Range = [Bound] | [Bound<MinComparator>, Bound<MaxComparator>]
 export interface RangeNode extends BaseNode<{ kind: "range"; rule: Range }> {
     min: Bound<MinComparator> | undefined
     max: Bound<MaxComparator> | undefined
+    numericMin: Bound<MinComparator, number | Date> | undefined
+    numericMax: Bound<MaxComparator, number | Date> | undefined
+    value: number | Date
 }
 
 export const rangeNode = defineNodeKind<RangeNode>(
     {
         kind: "range",
         parse: (input) => input,
         compile: (rule, ctx) => {
+            const l = rule[0].limit
```

Comment:
Can just assign using `.valueOf()` here in both cases and get rid of `numericL` and `numericR`